### PR TITLE
Add abap authorization double to "Exploit the test tools"

### PR DESCRIPTION
**Pull request by StefanButscher**
I have added a few lines and sample coding to show how to use the ABAP authorization double.

### CODE REVIEW

File: clean-abap/CleanABAP.md [hunk 1]
```diff
@@ -4606,6 +4606,15 @@ by redirecting them to test data defined in the unit test without interfering wi
 
 - Use the CDS test environment (`CL_CDS_TEST_ENVIRONMENT`) to test your CDS views.
 
+- Use the authorization double (`CL_AUNIT_AUTHORITY_CHECK`) to withdraw authorizations during the runtime of the test.
```
Comment (ConjuringCoffee) [inline clean-abap/CleanABAP.md:4609]: It's not an "authorization double" though, is it? The ABAP Doc of the class calls it an "AUTHORITY-CHECK test environment", so I think we should go with that.

Comment (ConjuringCoffee) [reply clean-abap/CleanABAP.md:4609]: Alternatively, "authority check test environment" would also work..

File: clean-abap/CleanABAP.md [hunk 2]
```diff
@@ -4606,6 +4606,15 @@ by redirecting them to test data defined in the unit test without interfering wi
...
+  This helps to test behavior of the code for different authorizations.
+
+```ABAP
+    DATA(auth_controller) = cl_aunit_authority_check=>get_controller( ).
+    DATA(auth_obj_set) = cl_aunit_authority_check=>create_auth_object_set(  user_role_authorizations ).
```
Comment (ConjuringCoffee) [inline clean-abap/CleanABAP.md:4614]: This conflicts with rule "Avoid abbreviations". I suggest using the variable names `authority_check_controller` and `authority_object_set` / `authorization_object_set`. 

I can understand wanting to abbreviate to `auth`, but I don't think we should abbreviate `objects`.

(Also, there's an unnecessary space before `user_role_authorizations`.)

**Comment by ConjuringCoffee** [comment]
The build step currently fails because some links from the review guide are dead now:

image

@larshp, is this a temporary error or should we adapt the links in a separate pull request?

File: clean-abap/CleanABAP.md (hunk 2 above)
Comment (StefanButscher) [reply clean-abap/CleanABAP.md:4614]: Valid point, I will look into this after my vacation.

**Comment by ConjuringCoffee** [comment]
I have fixed the links in #328.

**Review by ConjuringCoffee — APPROVED** [review]

**Comment by StefanButscher** [comment]
Can anyone review ?

File: clean-abap/CleanABAP.md [hunk 3]
```diff
@@ -4606,6 +4606,15 @@ by redirecting them to test data defined in the unit test without interfering wi
 
 - Use the CDS test environment (`CL_CDS_TEST_ENVIRONMENT`) to test your CDS views.
 
+- Use the AUTHORITY-CHECK test environment (`CL_AUNIT_AUTHORITY_CHECK`) to withdraw authorizations during the runtime of the test.
```
Comment (sautermi0) [inline clean-abap/CleanABAP.md:4609]: I'd rather call it a Helper API or something like that, not an environment... The official documentation calls it an API and not an environment: https://help.sap.com/docs/ABAP_PLATFORM_NEW/c238d694b825421f940829321ffa326a/6500d4d8f89a4743a6c0513d659a475b.html

File: clean-abap/CleanABAP.md [hunk 4]
```diff
@@ -4606,6 +4606,15 @@ by redirecting them to test data defined in the unit test without interfering wi
...
+  This helps to test behavior of the code for different authorizations.
+
+```ABAP
+    DATA(authority_check_controller) = cl_aunit_authority_check=>get_controller( ).
+    DATA(authority_object_set) = cl_aunit_authority_check=>create_auth_object_set( user_role_authorizations ).
```
Comment (sautermi0) [inline clean-abap/CleanABAP.md:4614]: I think an example of what the user_role_authorizations look like would be nice as well...

File: clean-abap/CleanABAP.md (hunk 3 above)
Comment (ConjuringCoffee) [reply clean-abap/CleanABAP.md:4609]: I agree with you, however I'd like to note that the ABAP Doc calls it a "Classic AUTHORITY-CHECK Test Environment" 😉

Comment (sautermi0) [reply clean-abap/CleanABAP.md:4609]: I just reached out to the developer responsible for the class and asked them what the best name would be and whether they could align documentation and the ABAP Doc ;)

File: clean-abap/CleanABAP.md (hunk 4 above)
Comment (StefanButscher) [reply clean-abap/CleanABAP.md:4614]: @sautermi0  I will add this in a future request. As of now I would like to complete the current one.

File: clean-abap/CleanABAP.md (hunk 3 above)
Comment (sautermi0) [reply clean-abap/CleanABAP.md:4609]: Just got the reply: It is supposed to be called a Test Helper API. The ABAP Doc will be adapted (I guess with 2311).

**Review by sautermi0 — APPROVED** [review]

**Review by N2oB6n-SAP — CHANGES_REQUESTED** [review]
I suggest a slight formatting change. Generally, I approve this addition.

**Comment by sautermi0** [comment]
I renamed the tool from an environment to a test helper API, included the requested formatting changes and added an example for user_role_authorizations.

@SAP/clean-abap-review-and-discussion-moderation Please check ;)

**Review by N2oB6n-SAP — APPROVED** [review]

**Review by cribuc — APPROVED** [review]

**Review by bjoern-jueliger-sap — APPROVED** [review]

**Pull request merged**